core/src/main/java/com/amplifyframework/core/model/ModelConverter.java

```diff
@@ -53,28 +55,54 @@ public static <T extends Model> Map<String, Object> toMap(T instance, ModelSchem
                 result.put(fieldName, extractFieldValue(modelField.getName(), instance, schema));
             } else if (association.isOwner()) {
                 ModelSchema nestedSchema = schemaRegistry.getModelSchemaForModelClass(targetType);
-                Object associateId = extractAssociateId(modelField, instance, schema);
+                Map<String, Object> associateId = extractAssociateId(modelField, instance, schema);
                 if (associateId == null) {
                     // Skip fields that are not set, so that they are not set to null in the request.
                     continue;
                 }
                 result.put(fieldName, SerializedModel.builder()
-                    .serializedData(Collections.singletonMap("id", associateId))
                     .modelSchema(nestedSchema)
+                    .serializedData(associateId)
                     .build());
             }
             // Ignore if field is associated, but is not a "belongsTo" relationship
         }
         return result;
     }
 
-    private static Object extractAssociateId(ModelField modelField, Model instance, ModelSchema schema)
+    private static Map<String, Object> extractAssociateId(ModelField modelField, Model instance,
+                                                                    ModelSchema schema)
             throws AmplifyException {
         final Object fieldValue = extractFieldValue(modelField.getName(), instance, schema);
         if (modelField.isModel() && fieldValue instanceof Model) {
-            return ((Model) fieldValue).getId();
+            Model associatedModel = (Model) fieldValue;
+            ModelSchema childSchema =
+                    SchemaRegistry.instance().getModelSchemaForModelClass(associatedModel.getModelName());
+            /* Loop through primary key fields and get their respective values from the key map.
+              Deserialize the key value to the model with primary key values populated.
+              This will be used to create appsync delete mutation.  */
+
+            HashMap<String, Object> hashMap = new HashMap<>();
+            if (childSchema.getPrimaryIndexFields().size() > 1 && (associatedModel.resolveIdentifier()
+                    instanceof ModelIdentifier)) {
```

Comment:
This condition resolves `false` always when the `associatedModel` is an instance of `SerializedModel`. 

Looking at [the implementation](https://github.com/aws-amplify/amplify-android/pull/1650/files#diff-67adc6a20a2b32f0d0662b0d736fc543e38cbb8c4c372d37d5b3bbbe10079af1R211-R213) of `resolveIdentifier` method on `SerializedModel`, it returns `modelId` as instance of `Serializable`. 

Looking at [the value assignment](https://github.com/aws-amplify/amplify-android/pull/1650/files#diff-67adc6a20a2b32f0d0662b0d736fc543e38cbb8c4c372d37d5b3bbbe10079af1R340) of `modelId`, it invokes function `getUniqueKey`, [its implementation](https://github.com/aws-amplify/amplify-android/pull/1650/files#diff-aa418156dfeb9c52a3e64b4f0cb96727b5801643e2ecaa3e6e7c3ad05f3cd7d0R89-R103) returns a `String`, which implements `Serializable`, but has no intersection with `ModelIdentifier`. Therefore, `associatedModel.resolveIdentifier() instanceof ModelIdentifier` always resolves `false` for `SerializedModel`.

Reply:
Resolved in a separate branch.